RUN-LOCALLY.md

```diff
@@ -0,0 +1,146 @@
+# Run Locally
+
+## Prerequisites
+
+Ensure you have the following installed:
+
+- **Docker Engine**: Version 18.02 or higher. Installation instructions can be found at https://docs.docker.com/engine/install/
+
+- **Docker Compose Plugin**: Ensure you have the Compose plugin installed:
+
+  ```bash
+  docker compose version
+  # Install with: sudo apt install docker-compose-plugin
+  ```
+
+- **Blockchain Wallet**: The ZNode will dispatch proofs to a blockchain contract, which requires a funded wallet account on the target blockchain (for IoTeX Testnet, see how to [create a wallet](https://docs.iotex.io/the-iotex-stack/wallets/metamask), and [claim test IOTX](https://docs.iotex.io/the-iotex-stack/iotx-faucets/testnet-tokens#the-iotex-developer-portal))
+
+- **Bonsai API Key**: If you plan to generate RISC0 snark proofs, as the ZNode protocol currently relies on the [Bonsai API](https://dev.risczero.com/api/bonsai/) you'll need to get [their API key](https://docs.google.com/forms/d/e/1FAIpQLSf9mu18V65862GS4PLYd7tFTEKrl90J5GTyzw_d14ASxrruFQ/viewform).
+
+## Get Repository
+```bash
+git clone https://github.com/machinefi/sprout.git
+cd sprout
+```
+
+## Generate ZKP
+
+### Compile the customized Halo2 circuit
+
+1. Install `wasm-pack`
+```bash
+curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+```
+
+2. Build wasm
+
+```bash
+cd examples/halo2-circuits/
+wasm-pack build --target nodejs --out-dir pkg
+```
+
+you will find `halo2_wasm_bg.wasm` under the `pkg` folder.
+
+3. (Optional) You can also write your circuit according to the [halo2 development documentation](https://zcash.github.io/halo2/user/simple-example.html), and put the circuit file in `src/circuits`; replace the `TODO` in `src/lib.rs` and build wasm with `wasm-pack build --target nodejs --out-dir pkg`.
+
+More details and options for `Halo2 circuit` are given in [its README](./examples/halo2-circuits/README.md).
+
+### Compile the customized Risc0 circuits
+
+1. Build
+
+```bash
+cd examples/risc0-circuits/
+cargo build --release
+```
+
+you will find `methods.rs` in the `target/release/build/risc0-circuits-xxx/out/methods.rs`
```

Comment:
change this to `The directory of methods.rs will be printed to the console, like this

warning: methods_path is: "sprout/examples/risc0-circuits/target/release/build/risc0-circuits-5efc4ff59af940ab/out/methods.rs"
`.  I print the path of `method.rs` on the console